src/transform/plugins/code.ts

```diff
@@ -68,10 +68,48 @@ function termReplace(str: string, env: EnvTerm, escape: (str: string) => string)
     return termCode || str;
 }
 
+function padNumber(num: number, width: number): string {
+    const numStr = String(num);
+    return ' '.repeat(Math.max(0, width - numStr.length)) + numStr;
+}
+
+function addLineNumbers(code: string): string {
+    const lines = code.split('\n');
+    const lineCount = lines.length;
+    const maxDigits = String(lineCount).length;
+
+    // Remove trailing empty line if it exists
+    const hasTrailingNewline = code.endsWith('\n');
+    const linesToProcess = hasTrailingNewline ? lines.slice(0, -1) : lines;
+
+    return (
+        linesToProcess
+            .map((line, index) => {
+                const lineNumber = padNumber(index + 1, maxDigits);
+                return `<span class="yfm-line-number">${lineNumber}</span>${line}`;
+            })
+            .join('\n') + (hasTrailingNewline ? '\n' : '')
+    );
+}
+
 const code: MarkdownItPluginCb = (md) => {
     const superCodeRenderer = md.renderer.rules.fence;
     md.renderer.rules.fence = function (tokens, idx, options, env, self) {
-        const superCode = superCodeRenderer?.(tokens, idx, options, env, self);
+        const token = tokens[idx];
```

Comment:
please add tests for generating line numbers in code blocks

https://github.com/diplodoc-platform/transform/tree/master/test